polus/plugins.py

```diff
@@ -559,10 +606,42 @@ def submit_plugin(manifest: typing.Union[str, dict, pathlib.Path]) -> Plugin:
         with open(org_path.joinpath(out_name), "w") as fw:
             json.dump(manifest, fw, indent=4)
 
+    # Refresh plugins list if refresh = True
+    if refresh:
+        plugins.refresh()
+
     # Return in case additional QA checks should be made
     return plugin
 
 
+def add_plugin(
+    user: str,
+    branch: str,
+    plugin: str,
+    repo: str = "polus-plugins",
+    manifest_name: str = "plugin.json",
+):
+    """Add plugin from GitHub.
+
+    This function adds a plugin hosted on GitHub and returns a Plugin object.
+
+    Args:
+        user: GitHub username
+        branch: GitHub branch
+        plugin: Plugin's name
+        repo: Name of GitHub repository, default is `polus-plugins`
+        manifest_name: Name of manifest file, default is `plugin.json`
+
+    Returns:
+        A Plugin object populated with information from the plugin manifest.
+    """
+    l = [user, repo, branch, plugin, manifest_name]
+    u = "/".join(l)
+    url = urljoin("https://raw.githubusercontent.com", u)
+    logger.info("Adding %s" % url)
+    return submit_plugin(url, refresh=True)
```

Comment:
Why not use the existing github auth function we currently have in place?

Reply:
These arguments are just to build a url